File: Python/백준/Bronze/2745. 진법 변환/진법 변환.py
import sys,math

input = sys.stdin.readline

b,n = map(str,input().split())
n = int(n)
m = len(b)-1
alpha = ['A','B','C','D','E','F','G','H','I','J','K','L','M','N','O','P','Q','R','S','T','U','V','W','X','Y','Z']

ans = 0
for i in range(m, -1 ,-1):
    k = 0
    sq = m -i
    if b[i] in alpha:
        k = alpha.index(b[i]) + 10
    else:
        k = int(b[i])
    ans += (math.pow(n,sq)*k)
    
print(int(ans))

File: C++17/백준/Bronze/2745. 진법 변환/진법 변환.cc
#include <iostream>
#include <string>
#include <math.h>

int num_convert(const char &str){
    if ('0' <= str && str <= '9'){
        return str - '0';
    }

    return str - 'A' + 10;
}

int main(void){
    std::string N;
    int B,size;
    std::cin >> N >> B;
    unsigned long long int ans = 0;
    
    size = N.length();

    for(int i = 0; i < size; i++){
        ans += std::pow(B,size -1 -i) * num_convert(N[i]);
    }
    
    std::cout << ans;
    return 0;
}